enhancements/storage/csi-azure-patches.md

```diff
@@ -0,0 +1,217 @@
+
+So far we've addressed this problem by directly patching the vendor
+directory (see
+[this](https://github.com/openshift/azure-disk-csi-driver/pull/60) and
+[this](https://github.com/openshift/azure-disk-csi-driver/pull/62)),
+however, this is not sustainable in the long term.
+
+### User Stories
+
+* As an OpenShift engineer,I want to fix issues in Azure CSI drivers
+without merging additional code, so that I can be sure I am not
+introducing other issues while doing so.
+
+### Goals
+
+
+The goal of this enhancement is to document a strategy for updating
+the `cloud-provider-azure` dependency within our CSI drivers' forks in
+older versions of OCP.
+
+### Non-Goals
+
+We will not cover other dependencies or forks.
+
+## Proposal
+
+We propose to have branches in `openshift/cloud-provider-azure`
+repository that only contains fixes for the Azure CSI drivers. Those
+branches are based off on the exact version that CSI drivers rely
+on. Additional bug fixes will be added on top of it.
+
+### Workflow Description
+
+The workflow is better described with an example. Consider the
+following problem:
+
+1. QE discovers an issue with the `azure-disk-csi-driver` in OCP 4.12,
+   with a bugfix being available in a newer `cloud-provider-azure`
+   dependency.
+1. In OCP 4.12, the driver depends on the version `v0.7.0` of
+   `cloud-provider-azure`.
+1. A bugfix is available in `v0.11.0`.
+1. Bumping the dependency to that version brings in several unrelated
+   code changes.
+
+Our proposed solution:
+
+1. In `openshift/cloud-provider-azure`, a Staff Engineer manually
+   creates a branch called `csi-4.12-patches` based on the upstream
+   `v0.7.0`.
+   1. This only needs to be done once per OCP version, on demand.
+1. The engineer working on the issue then creates a local branch off
+   of `csi-4.12-patches` and then applies the fix for the given issue
+   on top of it.
+1. The engineer runs some basic code verification and unit tests.
+1. The engineer submits a PR to `openshift/cloud-provider-azure`
+   targeting the `csi-4.12-patches` branch.
+1. Once the PR merges, the engineer can then bump the
+   `cloud-provider-azure` dependency in the `azure-disk-csi-driver`
+   repository using the [replace
+   directive](https://go.dev/ref/mod#go-mod-file-replace) in go.mod
+   file.
+
+### API Extensions
+
+NA.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+NA.
+
+#### Standalone Clusters
+
+NA.
+
+#### Single-node Deployments or MicroShift
+
+NA.
+
+### Implementation Details/Notes/Constraints
+
+NA.
+
+### Risks and Mitigations
+
+The biggest risk is to not have any CI in the CI branches in
+`openshift/cloud-provider-azure`. This means no automation (`\lgtm`)
+and no jobs. However, we expect to mitigate this risk by:
+
+1. Having CI in the `openshift/azure-disk-csi-driver` PR that will
+   bump the `cloud-provider-azure` dependency.
+1. Running tests locally before submitting the PR to
+   `openshift/cloud-provider-azure`.
```

Comment:
If we really want CI, we can always do a WIP PR in the CSI driver repo with manually patched vendor.